src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs

```diff
@@ -508,6 +509,43 @@ public float Cost
             if (missing.Length > 0)
                 throw _host.Except($"The following required inputs were not provided: {String.Join(", ", missing)}");
 
+            var inputInstance = _inputBuilder.GetInstance();
+            var warning = "Different {0} column specified in trainer and in macro: '{1}', '{2}'." +
+                        " Using column '{2}'. To column use '{1}' instead, please specify this name in" +
+                        "the trainer node arguments.";
+            if (!string.IsNullOrEmpty(label) && inputInstance is LearnerInputBaseWithLabel)
+            {
+                var labelInputInstance = inputInstance as LearnerInputBaseWithLabel;
+                if (label != labelInputInstance.LabelColumn)
+                    ch.Warning(warning, "label", label, labelInputInstance.LabelColumn);
+                else
+                    labelInputInstance.LabelColumn = label;
+            }
+            if (!string.IsNullOrEmpty(group) && inputInstance is LearnerInputBaseWithGroupId)
+            {
+                var groupInputInstance = inputInstance as LearnerInputBaseWithGroupId;
+                if (group != groupInputInstance.GroupIdColumn)
+                    ch.Warning(warning, "group Id", group, groupInputInstance.GroupIdColumn);
+                else
+                    groupInputInstance.GroupIdColumn = group;
+            }
+            if (!string.IsNullOrEmpty(weight) && inputInstance is LearnerInputBaseWithWeight)
+            {
+                var weightInputInstance = inputInstance as LearnerInputBaseWithWeight;
+                if (weight != weightInputInstance.WeightColumn)
+                    ch.Warning(warning, "weight", weight, weightInputInstance.WeightColumn);
+                else
+                    weightInputInstance.WeightColumn = weight;
+            }
+            if (!string.IsNullOrEmpty(weight) && inputInstance is UnsupervisedLearnerInputBaseWithWeight)
```

Comment:
>WithWeight [](start = 94, length = 10)

Is it possible to refactor it into IHaveLabelColumn, IHaveWeightColum, IHaveGroupColumn? And check for interface instead of abstract class?